update warnOldDocs.js and symlink for release

rkuhn opened the pull request.

rkuhn committed:
update warnOldDocs.js and symlink for release

This will also git commit before and after the upload of the docs, so as
to make sure that changes to warnOldDocs.js are recoverable.

patriknw commented:
LGTM I guess

rkuhn commented:
Don’t be so lax, JS needs review, too! :-)

ktoso commented on akka-docs/_sphinx/themes/akka/static/warnOldDocs.js:
```diff
@@ -3,28 +3,67 @@ jQuery(document).ready(function ($) {
-  $.get(versionsJsonUri(), function (akkaVersionsData) {
-    var thisVersion = browsedAkkaVersion();
-    if (thisVersion.includes("-SNAPSHOT")) {
+  $.get("//akka.io/versions.json", function (akkaVersionsData) {
```
oh, interesting (the `//`)

ktoso commented on project/scripts/release:
```diff
@@ -361,16 +361,13 @@ if [ $dry_run ]; then
-  important scp $script_dir/../../akka-docs/versions.json ${publish_path}/docs/
-fi
+echolog "*****"
+echolog "Do not forget to update versions.json on akka.github.com!"
```
Yeah, good

ktoso commented on akka-docs/_sphinx/themes/akka/static/warnOldDocs.js:
```diff
@@ -36,17 +75,25 @@ function showVersionWarning(version, akkaVersionsData, seriesInfo) {
+      type: 'HEAD',
+      success: function() {
+        $('#samePageLink').html('<a href="' + insteadPage + '">Click here to go to the same page on the ' + instead.latest + ' version of the docs.</a>');
+      }
```
great! :-)

ktoso commented on akka-docs/_sphinx/themes/akka/static/warnOldDocs.js:
```diff
@@ -3,28 +3,67 @@ jQuery(document).ready(function ($) {
+
+function targetUrl(samePage, site, instead) {
+  var page = site.r;
+  if (samePage !== true) {
```
have to share this wonderful javascript programming pattern: `if (!!!samePage)`

ktoso replied:
(keep as it, just sharing for fun (also works with undefined etc))

ktoso commented:
Very good, LGTM

rkuhn commented:
I added the warning for browsing the snapshot docs, recommending users to view `current` instead, and I also found and fixed a problem with cookie storage (since the same script is now used for both akka and akka-stream-and-http-experimental.
